source/common/uma_pools/disjoint_pool.hpp

```diff
@@ -0,0 +1,78 @@
+//===---------- disjoint_pool.hpp - Allocator for USM memory --------------===//
```

Comment:
Why disjoint_pool implementation is not part of the `unified_memory_allocation` directory? Is it just an intermediate step?

Reply:
As I mentioned offline, it makes more sense to keep it as part of UR rather than UMA for now, so that we can use UR utils (like logger) + it makes it clear that we can safely change the API (e.g. for config).